packages/hooks/src/useGrowthbookFeatureFlag.ts

```diff
@@ -1,28 +1,45 @@
 import { useState, useEffect } from 'react';
 import { Analytics } from '@deriv-com/analytics';
+import { useRemoteConfig } from '@deriv/api';
 
-interface UseGrowthbookFeatureFlagArgs<T> {
+interface UseGrowthbookFeatureFlagArgs {
     featureFlag: string;
-    defaultValue: T;
 }
 
-const useGrowthbookFeatureFlag = <T extends string | boolean>({
-    featureFlag,
-    defaultValue,
-}: UseGrowthbookFeatureFlagArgs<T>) => {
-    const [featureFlagValue, setFeatureFlagValue] = useState<T>(defaultValue);
+const useGrowthbookFeatureFlag = ({ featureFlag }: UseGrowthbookFeatureFlagArgs) => {
+    const [featureFlagValue, setFeatureFlagValue] = useState(Analytics?.getFeatureValue(featureFlag));
```

Comment:
```suggestion
    const [featureFlagValue, setFeatureFlagValue] = useState(null);
```

Let's change the initial value to `null` which is not `undefined`, this came to my mind after thinking about the use case evgeniy has mentioned.

The idea is, we will set the first/default value to `null` which is not `undefined` and when the growthbook sdk is loaded and we set the value on line no. 30 we will either get, `undefined` (if the feature flag is toggled off) or the actual value i.e `true`.

This way we can differentiate between a feature flag which is actually turned off after the growthbook sdk is loaded

Reply:
![Uploading image.png…]()

Reply:
I have doubts on this pattern amir
this can make alot of confusion.
we may need to have another state for that loaded, not loaded state

Reply:
I added another state isGBLoaded and returning it in the return statement of the hook. 
that is also a boolean, which makes it much more easier to understand.